service/src/LevelZero.cpp

```diff
@@ -267,6 +267,72 @@ namespace geopm
             }
         }
 
+        //Cache performance domains
```

Comment:
Suggestion: consider refactoring this function due to its size (for readability and maintainability). It seems like it could be broken down into smaller helper functions, one for each domain being cached.